test/unit/org/apache/cassandra/config/ConfigCompatabilityTest.java

```diff
@@ -0,0 +1,369 @@
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.node.TextNode;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import org.checkerframework.checker.units.qual.K;
+import org.yaml.snakeyaml.introspector.Property;
+
+/**
+ * To create the test files used by this class, run {@link org.apache.cassandra.distributed.upgrade.ConfigCompatabilityTestGenerate}.
+ */
+public class ConfigCompatabilityTest
+{
+    private static final Logger logger = LoggerFactory.getLogger(ConfigCompatabilityTest.class);
+
+    public static final String TEST_DIR = "test/data/config";
+
+    // see CASSANDRA-11115
+    private static final Set<String> THRIFT = ImmutableSet.of("rpc_server_type",
+                                                              "rpc_port",
+                                                              "rpc_listen_backlog",
+                                                              "start_rpc",
+                                                              "thrift_max_message_length_in_mb",
+                                                              "rpc_max_threads",
+                                                              "rpc_min_threads",
+                                                              "rpc_recv_buff_size_in_bytes",
+                                                              "rpc_send_buff_size_in_bytes",
+                                                              "thrift_framed_transport_size_in_mb",
+                                                              "thrift_prepared_statements_cache_size_mb",
+                                                              "request_scheduler",
+                                                              "request_scheduler_id",
+                                                              "request_scheduler_options");
+    // see // CASSANDRA-16956
+    private static final Set<String> WINDOWS = ImmutableSet.of("windows_timer_interval");
+    /**
+     * Breaking Change: These properties were dead code but removed in a minor (4.0.2).  This broke our deprecation rules
+     * as they should have been removed in 5.0 as previously agreed..
+     */
+    private static final Set<String> CASSANDRA_17132 = ImmutableSet.of("otc_coalescing_enough_coalesced_messages",
+                                                                       "otc_coalescing_window_us",
+                                                                       "otc_coalescing_strategy");
+    private static final Set<String> REMOVED_IN_40 = ImmutableSet.<String>builder()
+                                                     .addAll(THRIFT)
+                                                     .addAll(WINDOWS) // windows was removed later, but support was dropped in 4.0
```

Comment:
But according to the latest conversations we shouldn't break upgrades except in a major and we still plan 4.1 not 5 so the windows one was returned as a deprecated placeholder that does nothing, same as others we have... If something changes we can reconsider it